cxx/distributions/adapter_test.cc

```diff
@@ -12,16 +12,22 @@ namespace tt = boost::test_tools;
 
 BOOST_AUTO_TEST_CASE(adapt_normal) {
   std::mt19937 prng;
-  DistributionAdapter<double> ad(new Normal(&prng));
+  Normal* n = new Normal(&prng);
+  DistributionAdapter<double> ad(n);
 
   ad.incorporate("5.0");
   ad.incorporate("-2.0");
+  BOOST_TEST(n->N == 2);
```

Comment:
If we're going to use `DistributionAdapter` as a drop-in for the other cluster-model distributions, we should add `++N;` to `incorporate` (and `--N` to `unincorporate`) and check `ad.N` in the test here, due to this assertion: https://github.com/probcomp/hierarchical-irm/blob/master/cxx/hirm.hh#L529C9-L529C45 (`N` is initialized to 0 in the distribution base class). I thought I added `++N`/`--N` in a previous PR but I might have lost it in a merge. (As of now, we probably won't need `DistributionAdapter` in the future since it was an alternative to making `Relation` a templated class).

Reply:
Done.